src/client/web/WebRTC.ts

```diff
@@ -0,0 +1,114 @@
+import Peer from 'peerjs'
+import Network from '../services/Network'
+
+export default class WebRTC {
+  private myPeer: Peer
+  private peers = {}
+  private videoGrid = document.getElementById('video-grid')
+  private buttonGrid = document.getElementById('button-grid')
+  private myVideo = document.createElement('video')
+  private myStream?: MediaStream
+
+  constructor(userId: string, network: Network) {
+    this.myPeer = new Peer(userId)
+
+    // mute your own video stream (you don't want to hear to yourself)
+    this.myVideo.muted = true
+
+    // ask the browser to get use media
+    navigator.mediaDevices
+      .getUserMedia({
+        video: true,
+        audio: true,
+      })
+      .then((stream) => {
+        this.myStream = stream
+        this.addVideoStream(this.myVideo, this.myStream)
+
+        // prepare to be called
+        this.myPeer.on('call', (call) => {
+          call.answer(this.myStream)
+          const video = document.createElement('video')
+          call.on('stream', (userVideoStream) => {
+            this.addVideoStream(video, userVideoStream)
+          })
+          call.on('close', () => {
+            video.remove()
+          })
+
+          this.peers[call.peer] = call
+        })
+
+        this.setUpButtons()
+
+        network.readyToConnect()
+      })
+  }
+
+  // method to call a peer
+  connectToNewUser(userId: string) {
+    if (!this.myStream) return
+    const call = this.myPeer.call(userId, this.myStream)
+    const video = document.createElement('video')
+    call.on('stream', (userVideoStream) => {
+      this.addVideoStream(video, userVideoStream)
+    })
+    call.on('close', () => {
+      video.remove()
+    })
+
+    this.peers[userId] = call
+  }
+
+  // method to add new video stream to videoGrid div
+  addVideoStream(video: HTMLVideoElement, stream: MediaStream) {
+    video.srcObject = stream
+    video.addEventListener('loadedmetadata', () => {
+      video.play()
+    })
+    if (this.videoGrid) {
+      this.videoGrid.append(video)
+    }
+  }
+
+  // method to remove video stream a peer disconnects
+  deleteVideoStream(userId: string) {
+    if (this.peers[userId]) this.peers[userId].close()
+  }
+
+  // method to set up mute/unmute and video on/off buttons
+  setUpButtons() {
+    const audioButton = document.createElement('button')
+    audioButton.innerHTML = 'Mute'
```

Comment:
`innerHTML` is dangerous and can lead to XSS, use `innerText` instead since you're not using any HTML here.

Also there's many things I can pick here such as these event listeners are never cleaned up, but since we'll be transitioning these to react it doesn't matter...

Reply:
Hmm... I see. Changed it to ```innerText``` and look forward to React framework :D